File: app-miyou/src/main/java/com/miyou/app/domain/memory/port/MemoryRetrievalPort.java
package com.miyou.app.domain.memory.port;

import com.miyou.app.domain.dialogue.model.ConversationSessionId;
import com.miyou.app.domain.memory.model.MemoryRetrievalResult;
import reactor.core.publisher.Mono;

public interface MemoryRetrievalPort {
	Mono<MemoryRetrievalResult> retrieveMemories(ConversationSessionId sessionId,
		String query,
		int topK);
}

File: app-miyou/src/main/kotlin/com/miyou/app/domain/memory/port/MemoryRetrievalPort.kt
package com.miyou.app.domain.memory.port

import com.miyou.app.domain.dialogue.model.ConversationSessionId
import com.miyou.app.domain.memory.model.MemoryRetrievalResult
import reactor.core.publisher.Mono

interface MemoryRetrievalPort {
    fun retrieveMemories(
        sessionId: ConversationSessionId,
        query: String,
        topK: Int,
    ): Mono<MemoryRetrievalResult>
}
